internal/app/vodtester/continuous_vod_tester.go

```diff
@@ -0,0 +1,177 @@
+package vodtester
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/PagerDuty/go-pagerduty"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/stream-tester/internal/testers"
+	"github.com/livepeer/stream-tester/messenger"
+)
+
+type (
+	// IContinuousVodTester ...
+	IContinuousVodTester interface {
+		// Start start test. Blocks until error.
+		Start(fileName string, testDuration, pauseDuration, pauseBetweenTests time.Duration) error
+		Cancel()
+		Done() <-chan struct{}
+	}
+
+	ContinuousVodTesterOptions struct {
+		PagerDutyIntegrationKey string
+		PagerDutyComponent      string
+		PagerDutyLowUrgency     bool
+		VodTesterOptions
+	}
+
+	continuousVodTester struct {
+		ctx                     context.Context
+		cancel                  context.CancelFunc
+		host                    string // API host being tested
+		pagerDutyIntegrationKey string
+		pagerDutyComponent      string
+		pagerDutyLowUrgency     bool
+		vtOpts                  VodTesterOptions
+	}
+)
+
+// NewContinuousVodTester returns new object
+func NewContinuousVodTester(gctx context.Context, opts ContinuousVodTesterOptions) IContinuousVodTester {
+	ctx, cancel := context.WithCancel(gctx)
+	server := opts.API.GetServer()
+	u, _ := url.Parse(server)
+	cvt := &continuousVodTester{
+		ctx:                     ctx,
+		cancel:                  cancel,
+		host:                    u.Host,
+		pagerDutyIntegrationKey: opts.PagerDutyIntegrationKey,
+		pagerDutyComponent:      opts.PagerDutyComponent,
+		pagerDutyLowUrgency:     opts.PagerDutyLowUrgency,
+		vtOpts:                  opts.VodTesterOptions,
+	}
+	return cvt
+}
+
+func (cvt *continuousVodTester) Start(fileName string, testDuration, pauseDuration, pauseBetweenTests time.Duration) error {
+	messenger.SendMessage(fmt.Sprintf("Starting continuous vod test of %s", cvt.host))
+	try := 0
+	notRtmpTry := 0
+	maxTestDuration := 4*testDuration + pauseDuration + 15*time.Minute
```

Comment:
This is not valid for the VOD test, it was calculated for the livestram test which does 2 streams.

This should just be `testDuration` IMO (and so we don't need the intermediate variable), which is already a high value like `4m` right now.